tokio-util/src/task/join_map.rs

```diff
@@ -416,8 +416,8 @@ where
     ///  * `None` if the `JoinMap` is empty.
     ///
     /// [`tokio::select!`]: tokio::select
-    pub async fn join_one(&mut self) -> Option<(K, Result<V, JoinError>)> {
-        let (res, id) = match self.tasks.join_one_with_id().await {
+    pub async fn join_next(&mut self) -> Option<(K, Result<V, JoinError>)> {
```

Comment:
we may want to add a [`#[doc(alias = ...)]` attribute](https://doc.rust-lang.org/rustdoc/advanced-features.html#add-aliases-for-an-item-in-documentation-search)
```suggestion
    #[doc(alias = "join_one")]
    pub async fn join_next(&mut self) -> Option<(K, Result<V, JoinError>)> {
```
here so that this will show up in docs searches for "join_one". i _think_ we can do this even if we also apply alice's suggestion of adding a `#[doc(hidden)]` `join_one` method.